### ISSUE TITLE
RFC: Update Docs for REST API and DynamoDB Storage

### ISSUE THREAD

**Issue by mtliendo** [aws-amplify/docs#1776]
**Pages**: [CLI -> API (Rest)](https://docs.amplify.aws/cli/restapi) and [Libraries -> API (Rest)](https://docs.amplify.aws/lib/q/platform/js)

**Feedback**:

![Screen Shot 2020-05-16 at 2 08 09 AM](https://user-images.githubusercontent.com/5106417/82113241-2beb3480-971a-11ea-8e8a-89b679fa5ac9.png)

Currently the Amplify Framework documentation tends to favor GraphQL over REST API implementations in terms of content available. It would be nice if there was more documentation on how to setup various REST API's so that customer's could choose whatever fits best for their use case.

#### Solution

I'm proposing new content be added to show varying CLI paths, recipes for usage, and general authorization practices. This would be for both API in terms of API Gateway + Lambda as well as API Gateway + Lambda + DynamoDB.

#### Business Case

GraphQL does offer more robust solutions regarding the dataStore API and schema based design patterns, however by increasing the documentation surrounding more traditional API's, the barrier to entry would be lowered since REST endpoints are typically more understood than GraphQL SDL. Also, in addition to helping those using other serverless automation tools transition to Amplify, there would be a path for teams to start with a REST setup and slowly transition to a graphQL setup--perhaps with the help of the `@function` or `@http` directives.

---

#### Path Forward

I'm open to solutions on updating the docs, however I did want to point out, that I currently have a post that details [how to bring in existing resources](https://dev.to/mtliendo/aws-amplify-using-existing-auth-and-api-resources-30pg) for use within an Amplify project that could be ported over to the docs in some fashion, in addition to a [video tutorial](https://www.youtube.com/watch?v=A2LXm9TZJck&t=749s) that demonstrates how to create a REST API with dynamodb and call it on the frontend (currently no available in the docs).

So in short, a list of common REST topics would go a long way, and in addition to my own content, I'm hoping others would contribute their wants, ideas, and content as well.

**Comment by mtliendo** [comment]
https://github.com/aws-amplify/amplify-cli/issues/1157

**Comment by swaminator** [comment]
Thanks for the RFC @mtliendo. Could you suggest some topics in the two areas for CLI and Lib?

**Comment by swaminator** [comment]
@mtliendo any ideas here?

**Comment by mtliendo** [comment]
Thanks for keeping this open while I took a brief pause from OSS! I'm happy to circle back in a day or so!

**Comment by mtliendo** [comment]
#### Ideas for the Lib

* A clone of the [data access patterns](https://docs.amplify.aws/cli/graphql-transformer/dataaccess)
  * This would greatly help answer common dev questions regarding varying access points
  * This would also likely require the most time invested
* Working with other aws resources
  - one of the advantages of REST is how it can be integrated with other AWS resources via the SDK (cognito, ses, etc.)
  - this would be a great place to showcase common policy/iam credential handling for services that Amplify doesn't include b default.

#### Ideas for the CLI

* Generated code included in screenshot/examples.
  * It would be nice if after the CLI walkthrough, there was also a screenshot and paragraph that showed and discussed the generated code.
* A recipe section that shows how to order in which one may want to add resources.
  * If I want to create an app that uses s3 for images, dynamodb, and api gateway, what CLI path will get me there?
  * If in the above scenario, one were to go out of order, how could the application be updated so that Amplify could properly configure the resources?

There's are just a few to get the conversation going, but I'd be happy to followup with others.

**Comment by mtliendo** [comment]
Closing since I think the Amplify guides serve as a good place to enhance walkthrough-style documentation.
